app/views/NotificationPreferencesView/index.js

```diff
@@ -0,0 +1,263 @@
+
+import { SWITCH_TRACK_COLOR } from '../../constants/colors';
+import StatusBar from '../../containers/StatusBar';
+import ListItem from '../../containers/ListItem';
+import Separator from '../../containers/Separator';
+import I18n from '../../i18n';
+import scrollPersistTaps from '../../utils/scrollPersistTaps';
+import styles from './styles';
+import sharedStyles from '../Styles';
+import database from '../../lib/realm';
+import RocketChat from '../../lib/rocketchat';
+import log from '../../utils/log';
+
+const SectionTitle = React.memo(({ title }) => <Text style={styles.sectionTitle}>{title}</Text>);
+
+const SectionSeparator = React.memo(() => <View style={styles.sectionSeparatorBorder} />);
+
+const Info = React.memo(({ info }) => <Text style={styles.infoText}>{info}</Text>);
+
+SectionTitle.propTypes = {
+	title: PropTypes.string
+};
+
+Info.propTypes = {
+	info: PropTypes.string
+};
+
+const OPTIONS = {
+	desktopNotifications: [{
+		label: 'Default', value: 'default'
+	}, {
+		label: 'All Messages', value: 'all'
+	}, {
+		label: 'Mentions', value: 'mentions'
+	}, {
+		label: 'Nothing', value: 'nothing'
+	}],
+	audioNotifications: [{
+		label: 'Default', value: 'default'
+	}, {
+		label: 'All Messages', value: 'all'
+	}, {
+		label: 'Mentions', value: 'mentions'
+	}, {
+		label: 'Nothing', value: 'nothing'
+	}],
+	mobilePushNotifications: [{
+		label: 'Default', value: 'default'
+	}, {
+		label: 'All Messages', value: 'all'
+	}, {
+		label: 'Mentions', value: 'mentions'
+	}, {
+		label: 'Nothing', value: 'nothing'
+	}],
+	emailNotifications: [{
+		label: 'Default', value: 'default'
+	}, {
+		label: 'All Messages', value: 'all'
+	}, {
+		label: 'Mentions', value: 'mentions'
+	}, {
+		label: 'Nothing', value: 'nothing'
+	}],
+	desktopNotificationDuration: [{
+		label: 'Default', value: 0
+	}, {
+		label: '1 second', value: 1
+	}, {
+		label: '2 seconds', value: 2
+	}, {
+		label: '3 seconds', value: 3
+	}, {
+		label: '4 seconds', value: 4
+	}, {
+		label: '5 seconds', value: 5
+	}],
+	audioNotificationValue: [{
+		label: 'None', value: 'none None'
+	}, {
+		label: 'Default', value: '0 Default'
+	}, {
+		label: 'Beep', value: 'beep Beep'
+	}, {
+		label: 'Ding', value: 'ding Ding'
+	}, {
+		label: 'Chelle', value: 'chelle Chelle'
+	}, {
+		label: 'Droplet', value: 'droplet Droplet'
+	}, {
+		label: 'Highbell', value: 'highbell Highbell'
+	}, {
+		label: 'Seasons', value: 'seasons Seasons'
+	}]
+};
+
+export default class NotificationPreferencesView extends React.Component {
+	static navigationOptions = () => ({
+		title: I18n.t('Notification_Preferences')
+	})
+
+	static propTypes = {
+		navigation: PropTypes.object
+	}
+
+	constructor(props) {
+		super(props);
+		this.rid = props.navigation.getParam('rid');
+		this.rooms = database.objects('subscriptions').filtered('rid = $0', this.rid);
+		this.state = {
+			room: JSON.parse(JSON.stringify(this.rooms[0] || {}))
+		};
+	}
+
+	onValueChangeSwitch = async(key, value) => {
```

Comment:
95% of `onValueChangeSwitch` and `onValueChangePicker` are the same.
Can you abstract?

Reply:
No , problems comes beacause api uses 1 and 0 and true or false in string format , this is going to get ugly if we try to make them as one function